internal/proxy/ui_requests.go

```diff
@@ -359,3 +377,29 @@ func (p *Proxy) startJenkins(ns, clusterURL string) (state clients.PodState, cod
 	}
 	return state, code, nil
 }
+
+func checkIfSessionCookieIsValid(pci CacheItem, cookie *http.Cookie) (bool, error) {
+	jenkinsURL := fmt.Sprintf("%s://%s", pci.Scheme, pci.Route)
+
+	req, _ := http.NewRequest("GET", jenkinsURL, nil)
+	req.AddCookie(cookie)
+
+	c := http.DefaultClient
```

Comment:
do not use the default client rather, use the client in utils which sets a timeout